src/Donors/resources/utils.ts

```diff
@@ -0,0 +1,74 @@
+import {useEntityRecord} from '@wordpress/core-data';
+import {Donor} from '@givewp/donors/admin/components/types';
+import type {GiveDonorOptions} from '@givewp/donors/types';
+import apiFetch from '@wordpress/api-fetch';
+
+declare const window: {
+    GiveDonorOptions: GiveDonorOptions;
+} & Window;
+
+/**
+ * @unreleased
+ */
+export function useDonorEntityRecord(donorId?: number) {
+    const urlParams = new URLSearchParams(window.location.search);
+
+    const {
+        record: donor,
+        hasResolved,
+        save,
+        edit,
+    }: {
+        record: Donor;
+        hasResolved: boolean;
+        save: () => any;
+        edit: (data: Donor | Partial<Donor>) => void;
+    } = useEntityRecord('givewp', 'donor', donorId ?? urlParams.get('id'));
+
+    return {donor, hasResolved, save, edit};
+}
+
+/**
+ * @unreleased
+ */
+export function getDonorOptionsWindowData(): GiveDonorOptions {
+    return window.GiveDonorOptions;
+}
+
+export function handleTooltipDismiss(id: string) {
+    return apiFetch({
+        url: window.GiveDonorOptions.adminUrl + '/admin-ajax.php?action=' + id,
+        method: 'POST',
+    })
+}
+
+/**
+ * @unreleased
+ */
+export function amountFormatter(currency: Intl.NumberFormatOptions['currency'], options?: Intl.NumberFormatOptions): Intl.NumberFormat {
```

Comment:
Do we need this helper function in almost every domain? Can we extract this and put it in one place so that we don't have to write it multiple times.

Reply:
Yeah, absolutely. Here it is: 6c7d0dbc22eecc4978c13a0a15d90738e56d2f62